pkg/kubelet/cm/helpers_linux.go

```diff
@@ -167,6 +167,12 @@ func ResourceConfigForPod(pod *v1.Pod, enforceCPULimits bool, cpuPeriod uint64,
 	// determine the qos class
 	qosClass := v1qos.GetPodQOS(pod)
 
+	// quota is not capped if cpu manager policy is static
+	// and pod meets qualifications for static assignment
+	if cpuExperimentalManagerPolicyStatic && qosClass == v1.PodQOSGuaranteed && cpuRequests % MilliCPUToCPU == 0 {
```

Comment:
The condition for having pinned cpus is to use an integer value in the container's request. This condition would return true even for requests.cpu: 1.0, am I right?

I like it, because it is much simpler than my PR https://github.com/kubernetes/kubernetes/pull/117030 , however this was one of the review comments I had to resolve.

Reply:
yessir! because we convert to mc first then look at the def this will handle all of the following cases
```
request: 1
request: 1000m
request: 1.0
```

Reply:
```
package main

import (
  "fmt"
  v1 "k8s.io/api/core/v1"
  "k8s.io/apimachinery/pkg/api/resource"
)

func computeCpuResourceQuantity(cpu string) resource.Quantity {
  res := v1.ResourceList{}
  res[v1.ResourceCPU] = resource.MustParse(cpu)
  return res["cpu"]
}

func main() {
  i := computeCpuResourceQuantity("1")
  j := computeCpuResourceQuantity("1.0")
  k := computeCpuResourceQuantity("1000m")
  fmt.Println(i.MilliValue())
  fmt.Println(j.MilliValue())
  fmt.Println(k.MilliValue())

}

```

Reply:
But that is exactly what is wrong wrt to the cpu manager logic. Only containers with an integer value "1" are pinned. "1.0" will not get pinned cpus.

Or course as long as the reviewers say that is ok I am happy with the simplification it brings.

Reply:
this is why we should have consistent logic, best if in cpumanager (and we can totally do this) :)

Reply:
its easy enough to check the precision if we wish to disallow "1.0" and only allow "1". let me know what is desired!